packages/expo/src/Notifications.ts

```diff
@@ -244,13 +284,28 @@ export default {
     }
   },
 
+  async scheduleLocalNotificationWithMatchIOSAsync(notification: LocalNotification, options: DateMatchingTrigger) {
+    if (Platform.OS === 'android') {
+       throw new Error("This function is supported only on iOS.");
+    }
+    return ExponentNotifications.scheduleLocalNotification(notification, options);
+  },
+
+  async scheduleLocalNotificationWithTimeIntervalIOSAsync(notification: LocalNotification, options: IntervalTrigger) {
+    if (Platform.OS === 'android') {
+       throw new Error("This function is supported only on iOS.");
+    }
+    return ExponentNotifications.scheduleLocalNotificationWithTimeInterval(notification, options);
+  },
   /* Schedule a notification at a later date */
   async scheduleLocalNotificationAsync(
     notification: LocalNotification,
     options: {
       time?: Date | number;
       repeat?: 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
       intervalMs?: number;
+      dateMatchingTriggerIOS?: DateMatchingTrigger;
+      intervalTriggerIOS?: IntervalTrigger;
```

Comment:
I believe that intervalMs is something else because docs say "(Android only) intervalMs (optional) (number) -- Repeat interval in number of milliseconds" so probably user can pass time option different than intervalMs. I think that we can't easily provide that for iOS.